pkg/controller/cloud/node_controller.go

```diff
@@ -430,7 +434,7 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Co
 		}
 	}
 
-	nodeAddresses, err := getNodeAddressesByProviderIDOrName(ctx, instances, node)
+	nodeAddresses, err := getNodeAddressesByProviderIDOrName(ctx, instances, providerID, node.Name)
```

Comment:
I don't believe its guaranteed that the providerID field has been set. In that case we are going to have a problem with this.

Reply:
@cheftako , can you elaborate details? What would be a problem?

Reply:
providerID is only set if node.Spec.ProviderID is not set. So as example if we go through this method twice for the same node, the second time the method is called, node.Spec.ProviderID will be set. As a result providerID on the second pass will *not* have been set here. Now we will fail to look up the node addresses because we aren't using the correct providerID...

Reply:
Yeah, going through this multiple times should not happen after https://github.com/kubernetes/kubernetes/pull/85735 but its not impossible if the updating fails on all attempts

Reply:
@cheftako , with the latest force-push the `providerID` should be set to `node.Spec.ProviderID` when `node.Spec.ProviderID != nil`

Reply:
Looks good